uarray/backend.py

```diff
@@ -164,6 +165,12 @@ def replace_dispatchables(
                 )
 
         args, kwargs = argument_replacer(args, kwargs, tuple(replaced_args))
+
+        kwargs = {k: kwargs[k] for k in opts if k in kwargs}
```

Comment:
This doesn't work because opts contains `Parameter` objects and compares them to strings. Instead, I think `opts` should be `defaults.keys()`.

That being said, is it even necessary? Shouldn't the call to `argument_extractor` fail if there are invalid `kwargs`. At least it did in my case.

Reply:
`opts` was meant to include the list of optional arguments, so that argument replacers could be unified... For example, `all` has no `dtype` arg, but `sum` does. Otherwise it'd require separate argument replacers for both.

If you prefer the other method, I'm willing to revert this change.

Reply:
I've fixed this in fd30f61144a7149b844e6b617941873df3f40abe.